File: src/mysqlcookbook/recipes/api/06_conn_params/option_file.rb
#!/usr/bin/ruby
# option_file.rb: demonstrate DSN options for reading MySQL option files

require "mysql2"

begin
#@ _FRAG1_
  # look in user-specific option file owned by the current user
  client = Mysql2::Client.new(:default_file => "#{ENV['HOME']}/.my.cnf",
                              :database => "cookbook")
#@ _FRAG1_
  puts "Connected"
rescue Mysql2::Error => e
  puts "Cannot connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
else
  client.close()
  puts "Disconnected"
end

begin
#@ _FRAG2_
  # look in standard option files; use [cookbook] and [client] groups
  client = Mysql2::Client.new(:default_group => "cookbook",
                              :database => "cookbook")
#@ _FRAG2_
  puts "Connected"
rescue Mysql2::Error => e
  puts "Cannot connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
else
  client.close()
  puts "Disconnected"
end

File: src/mysqlcookbook/recipes/api/06_conn_params/option_file.pl
#!/usr/bin/perl
# option_file.pl: demonstrate DSN options for reading MySQL option files

use strict;
use warnings;
use DBI;

{ # begin scope

#@ _FRAG1_
my $conn_attrs = {PrintError => 0, RaiseError => 1, AutoCommit => 1};
# basic DSN
my $dsn = "DBI:mysql:database=cookbook";
# look in user-specific option file owned by the current user
$dsn .= ";mysql_read_default_file=$ENV{HOME}/.my.cnf";
my $dbh = DBI->connect ($dsn, undef, undef, $conn_attrs);
#@ _FRAG1_
print "Connected\n";
$dbh->disconnect ();
print "Disconnected\n";

} # end scope

{ # begin scope

#@ _FRAG2_
my $conn_attrs = {PrintError => 0, RaiseError => 1, AutoCommit => 1};
# basic DSN
my $dsn = "DBI:mysql:database=cookbook";
# look in standard option files; use [cookbook] and [client] groups
$dsn .= ";mysql_read_default_group=cookbook";
my $dbh = DBI->connect ($dsn, undef, undef, $conn_attrs);
#@ _FRAG2_
print "Connected\n";
$dbh->disconnect ();
print "Disconnected\n";

} # end scope
